src/generate.rs

```diff
@@ -849,9 +849,9 @@ pub fn fields(
                 access: f.access,
                 evs: &f.enumerated_values,
                 sc: Ident::new(&*sc),
-                mask: util::unsuffixed_or_bool((1 << width) - 1, width),
+                mask: util::hex_or_bool((((1 as u64) << width) - 1) as u32, width),
                 name: &f.name,
-                offset: util::unsuffixed(u64(f.bit_range.offset)),
+                offset: util::hex(f.bit_range.offset),
```

Comment:
This value will in the range 0..32. TBH, I personally would find it easier to read it in decimal since it appears in the RHS of expressions like `value << offset`.

Reply:
I agree, will change that back.

Reply:
@japaric Actually I can't change that back because `offset` now takes a Token while `util::unsuffixed()` produces a Lit. If you want I can add some more code to produce a decimal token to be fed into offset.

Reply:
Or, maybe, `unsuffixed` could return `Tokens`.

Reply:
Okay, it's done.